### PR TITLE
Stair rotation

### PR DESCRIPTION
**Pull request by VitaliyShar**
Tried to add proper stair rotation while standing on node edge holding shift.

### CODE REVIEW

File: mods/stairs/init.lua [hunk 1]
```diff
@@ -32,7 +32,15 @@ local function rotate_and_place(itemstack, placer, pointed_thing)
 	if placer then
 		local placer_pos = placer:get_pos()
 		if placer_pos then
-			param2 = minetest.dir_to_facedir(vector.subtract(p1, placer_pos))
+			local vec = vector.subtract(p1, placer_pos)
+			param2 = minetest.dir_to_facedir(vec)
+			if math.abs(vec.z) <= 0.5 and  math.abs(vec.x) <= 0.5 and vec.y < 0 and p0.y == p1.y then
+				if param2 == 0 then param2 = 2
```
Comment (appgurueu) [inline mods/stairs/init.lua:38]: This is mapping 0 -> 2, 1 -> 3, 2 -> 0 and 3 -> 1. Instead of using `if`-`elseif` here, shorten this to `param2 = (param2 + 2) % 4`

**Review by appgurueu — CHANGES_REQUESTED** [review]
Overall I'm lacking the rationale & explanatory comments: What is the current behavior and how is it broken? How does this improve upon the current behavior?

**Comment by VitaliyShar** [comment]
explanation

**Comment by VitaliyShar** [comment]
This is my first step in lua and github. Sorry, my english not very well.

File: mods/stairs/init.lua [hunk 2]
```diff
@@ -32,7 +32,11 @@ local function rotate_and_place(itemstack, placer, pointed_thing)
 	if placer then
 		local placer_pos = placer:get_pos()
 		if placer_pos then
-			param2 = minetest.dir_to_facedir(vector.subtract(p1, placer_pos))
+			local vec = vector.subtract(p1, placer_pos)
```
Comment (appgurueu) [inline mods/stairs/init.lua:35]: `placer_pos` is the feet position of the player; you should probably use the placer's look dir instead (`player:get_look_dir()`).

File: mods/stairs/init.lua [hunk 3]
```diff
@@ -32,7 +32,11 @@ local function rotate_and_place(itemstack, placer, pointed_thing)
...
+			param2 = minetest.dir_to_facedir(vec)
+			if math.abs(vec.z) <= 0.5 and  math.abs(vec.x) <= 0.5 and vec.y < 0 and p0.y == p1.y then
```
Comment (appgurueu) [inline mods/stairs/init.lua:37]: This checks that:

* `math.abs(vec.x) <= 0.5 and math.abs(vec.z) <= 0.5`: Player is placing the node where he is standing? (XZ rectangle collision)
* `vec.y < 0`: Player is looking down? (feet are above placed node?)
* `p0.y == p1.y`: Stair was not placed above or under a node

Please remove the space before `math.abs` *and* name your variables properly *and* add some comments.

Comment (VitaliyShar) [reply mods/stairs/init.lua]: Yes, all these statements are true.

**Review by appgurueu — APPROVED** [review]
Please use `snake_case`. Otherwise LGTM.

**Review by sfan5 — APPROVED** [review]
works
